hls-plugin-api/src/Ide/Types.hs

```diff
@@ -191,19 +192,64 @@ instance PluginMethod TextDocumentRangeFormatting where
   pluginEnabled _ pid conf = (PluginId $ formattingProvider conf) == pid
   combineResponses _ _ _ _ (x :| _) = x
 
+-- ---------------------------------------------------------------------
+
+-- | Notifications that can be handled by plugins.
+-- 'ExtraParams' captures any extra data the IDE passes to the handlers for this method
+class HasTracing (MessageParams m) => PluginNotification m where
```

Comment:
I don't think we really need this class. Isn't `HasTracing (MessageParams m)` sufficient?

Reply:
It doesn't really make sense to disable notifications for a particular plugin via the config I think. Its more of an internal implementation detail that shouldn't be visible to the user.

We could disable notifications from a plugin if the plugin itself is disabled.

Reply:
Isn't this class also used to restrict what `Method`s are allowed to be used in `IdeNotification`?

Reply:
Yes, but is there any reason to have such a restriction?

Reply:
I'll remove it